src/transformers/integrations/peft.py

```diff
@@ -312,14 +326,24 @@ def enable_adapters(self) -> None:
 
         for _, module in self.named_modules():
             if isinstance(module, BaseTunerLayer):
-                module.disable_adapters = False
+                # The recent version of PEFT need to call `enable_adapters` instead
+                if hasattr(module, "enable_adapters"):
+                    module.enable_adapters(enabled=True)
+                else:
+                    module.disable_adapters = False
 
-    def active_adapter(self) -> str:
+    def active_adapter(self, return_multi_adapters: bool = True) -> Union[str, List[str]]:
         """
         If you are not familiar with adapters and PEFT methods, we invite you to read more about them on the PEFT
         official documentation: https://huggingface.co/docs/peft
 
-        Gets the current active adapter of the model.
+        Gets the current active adapter of the model. In case of multi-adapter inference (combining multiple adapters
+        for inference) returns the list of active adapters so that users can deal with them accordingly.
+
+        Args:
+            return_multi_adapters (`bool`, *optional*, defaults to `True`):
+                Whether to return a list of active adapters or not. If `False`, only the first adapter is returned. If
+                `True`, returns the list of active adapters.
```

Comment:
Are we sure about this API? It would seem better to me to have `active_adapter` changed to `active_adapters` and to always default to returning the list of active adapters.

Reply:
BC would be handled by having another method `active_adapter` that would log a warning + call `active_adapters` under the hood